backend/dataall/modules/s3_datasets_shares/services/share_managers/s3_access_point_share_manager.py

```diff
@@ -194,101 +219,97 @@ def check_target_role_access_policy(self) -> None:
             f'arn:aws:s3:{self.dataset_region}:{self.dataset_account_id}:accesspoint/{self.access_point_name}/*',
         ]
 
-        version_id, policy_document = IAM.get_managed_policy_default_version(
-            self.target_environment.AwsAccountId, self.target_environment.region, share_resource_policy_name
-        )
-        logger.info(f'Policy... {policy_document}')
-
-        s3_statement_index = S3SharePolicyService._get_statement_by_sid(
-            policy_document, f'{IAM_S3_ACCESS_POINTS_STATEMENT_SID}S3'
-        )
-
-        if s3_statement_index is None:
-            logger.info(f'IAM Policy Statement {IAM_S3_ACCESS_POINTS_STATEMENT_SID}S3 does not exist')
+        if not S3SharePolicyService.check_if_sid_exists(
+            f'{IAM_S3_ACCESS_POINTS_STATEMENT_SID}S3', share_policy_service.total_s3_access_point_stmts
+        ):
+            logger.info(f'IAM Policy Statement with base Sid: {IAM_S3_ACCESS_POINTS_STATEMENT_SID}S3 does not exist')
```

Comment:
nit: can we change the logs a bit? At first I assumed we were checking only if `{IAM_S3_ACCESS_POINTS_STATEMENT_SID}S3` exists, but it turns out that we are checking if there is any sid that starts with `{IAM_S3_ACCESS_POINTS_STATEMENT_SID}S3`

Reply:
Similar to the log error in line 290

Reply:
Made changes to make it more clear